### PR TITLE
SYNCT-274 Create extension points that allow implementers control over the sync process through code

### PR DESCRIPTION
**Pull request by rstencel**
## Description of what I changed

Added javadoc to ClientHelper interface.

## Issue I worked on

see https://issues.openmrs.org/browse/SYNCT-274

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java [hunk 1]
```diff
@@ -8,27 +8,112 @@
...
+ * <h1>ClientHelper</h1>
+ * Describes required methods for alternative node-to-node communication clients.
+ *
+ * @see <a href="https://issues.openmrs.org/browse/SYNCT-274">SYNCT-274</a>
+ * @since 1.14.0
+ */
 public interface ClientHelper {
 
+	/**
+	 * <p>Returns new 'retrieve' request - in means of HTTP protocol it refers to GET request.</p>
```
Comment (wojtekbu) [inline api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java]:
```suggestion
	 * <p>Implemments creation of retrive request appropiate for specific Client. </p>
```

File: api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java [hunk 2]
```diff
@@ -8,27 +8,112 @@
 import java.net.URISyntaxException;
 import java.util.List;
 
+/**
+ * <h1>ClientHelper</h1>
+ * Describes required methods for alternative node-to-node communication clients.
```
Comment (wojtekbu) [inline api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java]:
```suggestion
 * <p>Adapts specific Client to be used by pull & push business logic.</p>
```

File: api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java [hunk 3]
```diff
@@ -8,27 +8,112 @@
...
+	 * <p>Returns new 'retrieve' request - in means of HTTP protocol it refers to GET request.</p>
+	 *
+	 * @param url represents URL of resource to be retrieved
+	 * @return returns new RequestEntity with 'retrieve' request
+	 * @throws URISyntaxException
+	 */
 	RequestEntity retrieveRequest(String url) throws URISyntaxException;
 
+	/**
+	 * <p>Returns new 'create' request - in means of HTTP protocol it refers to POST request.</p>
```
Comment (wojtekbu) [inline api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java]: as above

File: api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java [hunk 4]
```diff
@@ -8,27 +8,112 @@
...
+	 *
+	 * @param url represents URL of resource category, where an object will be created
+	 * @param object represents an object that will be sent
+	 * @return returns new RequestEntity with 'create' request
+	 * @throws URISyntaxException
+	 */
 	RequestEntity createRequest(String url, Object object) throws URISyntaxException;
 
+	/**
+	 * <p>Returns new 'delete' request - in means of HTTP protocol it refers to DELETE request.</p>
```
Comment (wojtekbu) [inline api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java]: As above

File: api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java [hunk 5]
```diff
@@ -8,27 +8,112 @@
...
+	 *
+	 * @param url represents URL of resource category, from where an object will be deleted
+	 * @param uuid represents UUID of the object, that will be deleted
+	 * @return returns new RequestEntity with 'delete' request
+	 * @throws URISyntaxException
+	 */
 	RequestEntity deleteRequest(String url, String uuid) throws URISyntaxException;
 
+	/**
+	 * <p>Returns new 'update' request - in means of HTTP protocol it refers to PUT request.</p>
```
Comment (wojtekbu) [inline api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java]: As above

File: api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java [hunk 6]
```diff
@@ -8,27 +8,112 @@
...
+	/**
+	 * <p>Returns Class object corresponding to category name.</p>
+	 *
+	 * @param category represents name of category
+	 * @return returns Class object
+	 */
 	Class resolveClassByCategory(String category);
 
+	/**
+	 * <p>Returns a list of HTTP interceptors used in communication between nodes, including authentication of a user.</p>
```
Comment (wojtekbu) [inline api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java]:
```suggestion
	 * <p>Returns a list of HTTP interceptors used in communication between node.</p>
```

Comment (wojtekbu) [reply api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java]: Not necessary.

File: api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java (hunk 2 above)
Comment (rstencel) [reply api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java]: Changed!

File: api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java (hunk 1 above)
Comment (rstencel) [reply api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java]: Ok!

File: api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java (hunk 4 above)
Comment (rstencel) [reply api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java]: Changed!

File: api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java (hunk 5 above)
Comment (rstencel) [reply api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java]: Changed!

File: api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java (hunk 6 above)
Comment (rstencel) [reply api/src/main/java/org/openmrs/module/fhir/api/helper/ClientHelper.java]: Ok!

**Pull request merged**